client/examples/directory/directory_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/etcd/client"
+)
+
+func TestMain(m *testing.M) {
+	main()
+	code := m.Run()
+
+	//Clean
+	log.Println("Cleaning test...")
+
+	//Clean nodes within "myNodes
+	cfg := client.Config{
```

Comment:
This isn't good, we can't expect there to be an etcd running and do arbitrary things to it. Can you instead use [integration.NewCluster](https://github.com/coreos/etcd/blob/master/integration/cluster_test.go#L393), as [Xiang suggested](https://github.com/coreos/etcd/pull/3985#issuecomment-164506812)?